platform/darwin/src/MGLLoggingConfiguration_Private.h

```diff
@@ -11,6 +15,13 @@
         [[MGLLoggingConfiguration sharedConfiguration] logCallingFunction:function functionLine:line messageType:type format:(message), ##__VA_ARGS__]; \
     } \
 }
+#define MGLAssert(expression, message, ...) \
```

Comment:
Is the intention to stop using `NSAssert()` throughout the codebase? If so, should we add a check to fail the nitpick build if `NSAssert()` is used directly?

Reply:
No, there are parts for example that we use `NSLog` because it is required as a warning to the console, can't think of a reason to be more strict and constraint `NSAssert`.

Reply:
How would a contributor to this codebase decide between `NSAssert()` and `MGLAssert()`? Perhaps briefly mention the criteria in the contributing guide, as a flexible way of maintaining consistency.

Reply:
I'm thinking that perhaps we should use `MGLAssert` throughout the code base for consistency. Perhaps renaming the existing `MGLAssert` to something like `MGLLoggingAssert` would offer more clarity.